src/networking.c

```diff
@@ -855,7 +856,7 @@ void addReplyDouble(client *c, double d) {
              * but still avoid an extra memcpy of the whole number, we reserve space
              * for maximum header `$0000\r\n`, print double, add the resp header in
              * front of it, and then send the buffer with the right `start` offset. */
-            int dlen = snprintf(dbuf+7,sizeof(dbuf) - 7,"%.17g",d);
```

Comment:
@filipecosta90 is fpconv_dtoa equivalent to `%17g` in terms of precision, and maximum length? i.e. our `MAX_LONG_DOUBLE_CHARS`

what about `%17f`?

Reply:
what about the other places where we use `%.17g`?

Reply:
WRT:
- is fpconv_dtoa equivalent to %.17g - yes. Assumes fp is an IEEE 64-bit floating point number and It produces an IEEE double string representation. The invariant should be: "If an IEEE 754 double-precision number is converted to a decimal string with at least 17 significant digits, and then converted back to double-precision representation, the final result must match the original number.". 
- in terms of precision, and maximum length?. Yep. maximum length will never exceed 24 characters
- what about %.17f? I can't find any usage of it in redis code

Reply:
WRT: what about the other places where we use %.17g? 

I've proceed with all the changes that rely uppong %.17g conversion. Notice now this PR will also improve ZADD performance. I've add more testcases to prove it. Will produce a comparison table tomorrow.

Reply:
sorry, there's a `%.17lf` (not `%.17f`), and there's also a `%.16lg`.

so the ones you replaced are just the `%.17g` (not with `f` and not with `l` suffixes.

maybe the comment next to `MAX_D2STRING_CHARS` needs to be improved.

Reply:
added note to MAX_D2STRING_CHARS